org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/ProjectUtils.java

```diff
@@ -536,4 +540,41 @@ public static void removeJavaNatureAndBuilder(IProject project, IProgressMonitor
 		}
 	}
 
+	public static int getMaxProjectProblemSeverity() {
+		int maxSeverity = IMarker.SEVERITY_INFO;
+		for (IProject project : ProjectUtils.getAllProjects()) {
+			if (ProjectsManager.DEFAULT_PROJECT_NAME.equals(project.getName())) {
+				continue;
+			}
+			
+			try {
+				maxSeverity = Math.max(maxSeverity, project.findMaxProblemSeverity(null, true, IResource.DEPTH_ZERO));
+				if (ProjectUtils.isMavenProject(project)) {
+					IFile configFile = project.getFile(MavenProjectImporter.POM_FILE);
+					maxSeverity = Math.max(maxSeverity, configFile.findMaxProblemSeverity(null, true, IResource.DEPTH_ZERO));
+				} else if (ProjectUtils.isGradleProject(project)) {
+					List<String> gradleConfigs = Arrays.asList(
+						GradleProjectImporter.BUILD_GRADLE_DESCRIPTOR,
+						GradleProjectImporter.BUILD_GRADLE_KTS_DESCRIPTOR,
+						GradleProjectImporter.SETTINGS_GRADLE_DESCRIPTOR,
+						GradleProjectImporter.SETTINGS_GRADLE_KTS_DESCRIPTOR
+					);
+
+					for (String fileName : gradleConfigs) {
+						IFile configFile = project.getFile(fileName);
+						maxSeverity = Math.max(maxSeverity, configFile.findMaxProblemSeverity(null, true, IResource.DEPTH_ZERO));
```

Comment:
If the configFile doesn't exist, what happens with `configFile.findMaxProblemSeverity()`?

Reply:
Oh yes, I should check the existence